Added packages to list of contributors.

peterfabian opened the pull request:
Include all the contributors, also to sub packages in the contributor list. 
Also, changed the format to HTML to make it easier to use in block editor and switched to use the default branch, for now, and for simplicity.

To test: 

1. create a GH oAuth token via https://github.com/settings/tokens, it needs access to the repo
2. run `grunt contributors`
3. add date (e.g. 2020-03-10)
4. add GH token
5. Nonempty output should be generated into contributors.html

peterfabian committed:
Added packages to list of contributors.

peterfabian committed:
Renamed the file to produce correct filename.

claudiosanches commented on Gruntfile.js:
```diff
@@ -215,11 +215,33 @@ module.exports = function( grunt ) {
+					' --authToken <%= authToken %> --cols 6 --sortBy contributions --format html --sortOrder desc' +
+					' --showlogin true --filter "renovate-bot,apps/renovate,renovate,renovate[bot]" >> contributors.html',
+					'echo "Output generated to contributors.html."',
+				].join('&&')
```
Coding standards requires spaces in `.join( '&&' )`

claudiosanches commented on Gruntfile.js:
```diff
@@ -215,11 +215,33 @@ module.exports = function( grunt ) {
+					'./node_modules/.bin/githubcontrib --owner woocommerce --repo woocommerce-rest-api --fromDate <%= fromDate %>' +
+					' --authToken <%= authToken %> --cols 6 --sortBy contributions --format html --sortOrder desc' +
+					' --showlogin true --filter "renovate-bot,apps/renovate,renovate,renovate[bot]" >> contributors.html',
+					'echo "<h2>WooCommerce core</h2>" >> contributors.html',
```
I think we should move core to the first, like we does with the changelogs.

peterfabian replied:
I wanted to politely give credit to our guests first, but we can swap, sure.

claudiosanches reviewed (changes requested):
It works fine, but feels weird having this script here, currently some grunt packages are installed only to allow run this script once we release something.
We could move it to a new repository and create a proper tool for it, so other teams can use too.
Or at least move it to the `bin/` directory, so `grunt-prompt` can get removed, and also `grunt-shell`, since we run e2e tests now by the `package.json` file, those script we have on Grunt can get removed too.

But if we plan to keep on core, at least we should include an entry about `contributors.html` in the `.gitignore`.

contributors.html

peterfabian committed:
coding standards fix

peterfabian committed:
WC core first

peterfabian committed:
Add the contributors.html to gitignore.

Leaving the contributors.md there, as the contributors file was previously generated as .md and might still be around in some directories, etc.

peterfabian committed:
Moved the contributors script from Grunt to bash.

peterfabian committed:
Removed the old way to run e2e tests.

peterfabian commented:
> It works fine, but feels weird having this script here, currently some grunt packages are installed only to allow run this script once we release something.
We could move it to a new repository and create a proper tool for it, so other teams can use too.

Fair enough, but this is basically just a call of a npm package, so I'm not sure it deserves its own repository. It's true, though, that in some way this could be included in an 'ideal repo' implementation we will create.

> Or at least move it to the bin/ directory, so grunt-prompt can get removed, and also grunt-shell, since we run e2e tests now by the package.json file, those script we have on Grunt can get removed too.

Do you mean creating a bash script and placing it into bin folder instead of running the command as a grunt script? Sounds like a good idea, as grunt anyway just runs shell commands. Please check the update.

> But if we plan to keep on core, at least we should include an entry about contributors.html in the .gitignore.

Done that, thanks for noticing.

peterfabian committed:
Removed grunt-prompt and grunt-shell packages.

claudiosanches reviewed (approved):
Great work!
